data_safe_haven/functions/validators.py

```diff
@@ -45,10 +45,99 @@ def validate_ip_address(
         if ip_address:
             return str(ipaddress.ip_network(ip_address))
         return None
-    except:
+    except Exception:
         raise typer.BadParameter("Expected valid IPv4 address, for example '1.1.1.1'")
 
 
+def validate_azure_location(location: Optional[str]) -> Optional[str]:
+    if not location:
+        return None
+    if location in (
+        "asia",
+        "asiapacific",
+        "australia",
+        "australiacentral",
+        "australiacentral2",
+        "australiaeast",
+        "australiasoutheast",
```

Comment:
Is there an Azure python module way to do this? Not sure we want to maintain this list.

Reply:
There is, but then you add a dependency on Azure API which is going to complicate the dependency chain for something you're mainly using to verify CLI arguments. I'll add a method to Azure API for this.

Reply:
I've added `AzureAPI.get_locations()` but in order to construct `AzureAPI` you need to know `subscription_name`. That means that we can't call anything that depends on `AzureAPI.get_locations()` in `ConfigSectionAzure` since `subscription_name` is not known here (and may not be set)

Reply:
I've replace this with a minimal regex: `at least one lowercase letter followed by 0 or 1 number then 0 or more lowercase letters`.